### ISSUE TITLE
Unable to find the wrapper "https"

### ISSUE THREAD

**Issue by tylerlwsmith** [asdf-community/asdf-php#163]
### Provide environment information

```bash
OS:
Darwin My-MacBook-Pro.local 23.1.0 Darwin Kernel Version 23.1.0: Mon Oct  9 21:33:00 PDT 2023; root:xnu-10002.41.9~7/RELEASE_ARM64_T6031 arm64

SHELL:
zsh 5.9 (x86_64-apple-darwin23.0)

BASH VERSION:
3.2.57(1)-release

ASDF VERSION:
v0.13.1-0586b37

ASDF INTERNAL VARIABLES:
ASDF_DEFAULT_TOOL_VERSIONS_FILENAME=.tool-versions
ASDF_DATA_DIR=/Users/tyler/.asdf
ASDF_DIR=/Users/tyler/.asdf
ASDF_CONFIG_FILE=/Users/tyler/.asdfrc

ASDF INSTALLED PLUGINS:
elixir                       https://github.com/asdf-vm/asdf-elixir.git master a4c42e1
erlang                       https://github.com/asdf-vm/asdf-erlang.git master aaa3ee7
nodejs                       https://github.com/asdf-vm/asdf-nodejs.git master c5b7c40
php                          https://github.com/asdf-community/asdf-php.git master 1eaf4de
ruby                         https://github.com/asdf-vm/asdf-ruby.git master b3d44de
```

### To Reproduce

Run `asdf install php 8.3.1` or `PHP_WITHOUT_PEAR=yes asdf install php 8.3.1`.

### Describe the Bug

Installation appears to error out when trying to install `pear` or `composer` because https is not enabled in `php.ini` during the installation.

When running `asdf install php 8.3.1`, I get the following errors:

```txt
Warning: fopen(): Unable to find the wrapper "https" - did you forget to enable it when you configured PHP? in /private/var/folders/5_/jsnsc2s51wl08b_htp4xrctr0000gn/T/php-src-php-8.3.1/pear/fetch.php on line 66

Warning: fopen(https://pear.php.net/install-pear-nozlib.phar): Failed to open stream: No such file or directory in /private/var/folders/5_/jsnsc2s51wl08b_htp4xrctr0000gn/T/php-src-php-8.3.1/pear/fetch.php on line 66

Error..
fopen(https://pear.php.net/install-pear-nozlib.phar): Failed to open stream: No such file or directory
make: *** [install-pear] Error 1
```

When running `PHP_WITHOUT_PEAR=yes asdf install php 8.3.1`, I get the following errors:

```txt
Warning: copy(): Unable to find the wrapper "https" - did you forget to enable it when you configured PHP? in Command line code on line 1

Warning: copy(): Unable to find the wrapper "https" - did you forget to enable it when you configured PHP? in Command line code on line 1

Warning: copy(): Unable to find the wrapper "https" - did you forget to enable it when you configured PHP? in Command line code on line 1

Warning: copy(https://getcomposer.org/installer): Failed to open stream: No such file or directory in Command line code on line 1

Warning: hash_file(composer-setup.php): Failed to open stream: No such file or directory in Command line code on line 1
Installer corrupt
Warning: unlink(composer-setup.php): No such file or directory in Command line code on line 1

Could not open input file: composer-setup.php
tyler@My-MacBook-Pro installs %
```

### Expected Behaviour

PHP successfully installs.

**Comment by tylerlwsmith** [comment]
The issue resolves itself when I both instances of `homebrew_package_path openssl@1.1` to `homebrew_package_path openssl` in the `install` file.

**Comment by watsonarw** [comment]
@tylerlwsmith Thank you for the workaround! I was pulling my hair out wondering why it wasn't working on my newer machine, but was fine on an older one.

It seems like `openssl@1.1` is an undocumented dependency on MacOS, it might be helpful if it were explicitly documented/installed in the Github actions workflow.

**Comment by yoonwaiyan** [comment]
I changed the `install` file to point to `openssl` without specifying version 1.1 and it worked.

**Comment by TangRufus** [comment]
Continue @tylerlwsmith's awesome workaround but without editing files:

```sh-session
$ brew install openssl@1.1

## If you have openssl v3 installed (you most like do)
$ brew unlink openssl@3
Unlinking /opt/homebrew/Cellar/openssl@3/3.2.1... 5867 symlinks removed.

$ brew link openssl@1.1
Linking /opt/homebrew/Cellar/openssl@1.1/1.1.1w... 3999 symlinks created.

If you need to have this software first in your PATH instead consider running:
  echo 'export PATH="/opt/homebrew/opt/openssl@1.1/bin:$PATH"' >> ~/.zshrc

$ openssl version
OpenSSL 1.1.1w  11 Sep 2023

$ asdf install php xxx

## Restore openssl version to v3
$ brew unlink openssl@1.1
Unlinking /opt/homebrew/Cellar/openssl@1.1/1.1.1w... 3999 symlinks removed.

$ brew link openssl@3
Linking /opt/homebrew/Cellar/openssl@3/3.2.1... 5867 symlinks created.

$ openssl version
OpenSSL 3.2.1 30 Jan 2024 (Library: OpenSSL 3.2.1 30 Jan 2024)
```

**Comment by tristanjahier** [comment]
Oddly enough, after installing OpenSSL 1.1 via Homebrew and installing PHP 8.3, I did not need to switch to OpenSSL 1.1 to install other versions of PHP. I even tried to re-install PHP 8.3 with OpenSSL 3 linked and it worked properly. The build output shows that it actually uses `openssl@1.1`. Maybe we just need to have `openssl@1.1` installed?

**Comment by cameronmurphy** [comment]
OpenSSL 1.1 reached end of life on September 11, 2023, and as such I get this warning when installing `openssl@1.1` from Homebrew:
```
Warning: openssl@1.1 has been deprecated because it is not supported upstream!
```

Probably naive question: PHP 8.1 added support for OpenSSL 3.0 - can compile against that instead?

**Comment by godbout** [comment]
for those on macOS who want `openssl 3` with PHP:
* for PHP8.1+ edit the `install` file and replace `openssl@1.1` by `openssl@3`
* for PHP versions before 8.1, well, PHP only supports openssl@1.1 so you don't need to edit the `install` file, but you need to `brew install openssl@1.1`

**Comment by orthodoX** [comment]
> Oddly enough, after installing OpenSSL 1.1 via Homebrew and installing PHP 8.3, I did not need to switch to OpenSSL 1.1 to install other versions of PHP. I even tried to re-install PHP 8.3 with OpenSSL 3 linked and it worked properly. The build output shows that it actually uses `openssl@1.1`. Maybe we just need to have `openssl@1.1` installed?

You are correct. I just installed `openssl@1.1` and retried the installation and it worked, no need to switch versions or meddle with installation files

**Comment by matteeyah** [comment]
To expand on @jhclouse's solution - this is the incantation that worked for me `PHP_CONFIGURE_OPTIONS="--with-openssl=$(brew --prefix openssl) --with-iconv=$(brew --prefix libiconv)" asdf install php`

IMHO it's much better and safer to just point the install script to the right location of `openssl` instead of forcing it to use an outdated, unsupported version of `openssl`.

**Comment by orthodoX** [comment]
> To expand on [@jhclouse](https://github.com/jhclouse)'s solution - this is the incantation that worked for me `PHP_CONFIGURE_OPTIONS="--with-openssl=$(brew --prefix openssl) --with-iconv=$(brew --prefix libiconv)" asdf install php`
> 
> IMHO it's much better and safer to just point the install script to the right location of `openssl` instead of forcing it to use an outdated, unsupported version of `openssl`.

You're absolutely right 🚀 And also `openssl@1.1` is so outdated that it's not even available:

```sh 
Error: openssl@1.1 has been disabled because it is not supported upstream! It was disabled on 2024-10-24.
```

**Comment by uenokazuma** [comment]
It’s quite funny. I’ve been using AI for the past year and tried to resolve the same issue mentioned here using ChatGPT, Claude, and Gemini, but with no luck. I worked on it for several other days whenever I had some free time, then really focused on it for the last two days.

At that point, I was done with AI and decided to search whether someone else had the same issue. I found this, then tried to run:

`brew install openssl@1.1`
then
`PHP_WITHOUT_PEAR=yes asdf install php 8.2.29`

And I finally got this message:

```
Installer verified
All settings correct for using Composer
```

I didn’t export any PHP_* variables in .zshrc. I only used pkg-config and set PATH to the newer curl from Homebrew.

I checked which OpenSSL it was using, and got:
```
~ % which openssl
/opt/homebrew/bin/openssl
~ % openssl --version
OpenSSL 3.6.1 27 Jan 2026 (Library: OpenSSL 3.6.1 27 Jan 2026)
```

I don’t know why, but it looks like it still needs openssl@1.1 installed on the machine.

OS:
macOS Tahoe 26.2
Darwin MacBook-Pro 25.2.0 Darwin Kernel Version 25.2.0: Tue Nov 18 21:09:55 PST 2025; root:xnu-12377.61.12~1/RELEASE_ARM64_T8103 arm64
SHELL:
zsh 5.9 (arm64-apple-darwin25.0)
asdf version:
asdf version 0.18.0

**Comment by watsonarw** [comment]
> I don’t know why, but it looks like it still needs openssl@1.1 installed on the machine.

Why is because the [install script explicitly references openssl@1.1](https://github.com/asdf-community/asdf-php/blob/master/bin/install#L31). As for _why_ it references that, I'm not sure, as others have said, changing that line to a newer version of openssl doesn't seem to impact the ability to install php, so it might not need that specific version.
